File: solutins-using-java-python/src/main/app/com/leetcode/Array/1207_unique-number-of-occurrences.java
package com.leetcode.Array;

import java.util.HashMap;
import java.util.HashSet;

class Solution1207 {
    boolean uniqueOccurrences(int[] arr) {
        var hashMap = new HashMap<Integer, Integer>();
        for(int val: arr) {
            hashMap.put(val, hashMap.getOrDefault(val, 0) + 1);
        }

        var hashSet = new HashSet<Integer>(hashMap.values());
        return hashSet.size() == hashMap.size();
    }
}

File: src/main/app/com/leetcode/Array/1207_unique-number-of-occurrences.kt
package com.leetcode.Array

class `1207_unique-number-of-occurrences` {
    fun uniqueOccurrences(arr: IntArray): Boolean {
        val hashMap: HashMap<Int, Int> = HashMap()
        for (value in arr) {
            hashMap[value] = hashMap[value] ?: 0
            hashMap[value] = hashMap[value]!! + 1
        }

        val hashSet: HashSet<Int> = HashSet()
        for (pair in hashMap) {
            hashSet.add(pair.value)
        }
        return hashSet.size == hashMap.size
    }
}